torch/csrc/autograd/function.h

```diff
@@ -369,6 +370,12 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
   /// Returns the name of the dynamic type of the function, for debugging.
   virtual std::string name() const;
 
+  /// The difference between functions `should_compute_output` and
+  /// `task_should_compute_output`: `should_compute_output` is called
+  /// when constructing the node because the node itself is irreverent to the
```

Comment:
Not sure to follow what it means that the node itself is irrelevant?
Maybe just say that the first one is only considering the links that are in the actual graph while the second (that should only be called during the backward pass) takes also into account the current backward that is running to trim even more the outputs we need to compute.

Reply:
Thanks, that is more clear. Rephrased.